### DISCUSSION TITLE
how to add listener for keyborad?

### DISCUSSION THREAD

**Discussion by rabbitLove520** [enable3d/enable3d#128] [Help / Questions] [answered]
I have found some code from the [first-phaser-game-3d-version](https://enable3d.io/examples/first-phaser-game-3d-version.html):
```
// add keys
this.keys = {
    w: this.input.keyboard.addKey('w'),
    a: this.input.keyboard.addKey('a'),
    d: this.input.keyboard.addKey('d')
}
```
And How can I use `this.input` in my class?

**Comment by yandeu** [comment, marked as answer]
I you use Phaser, see http://phaser.io/examples/v3/category/input

else try  
https://github.com/yandeu/keyboard and https://github.com/yandeu/tap
